src/tempdir.cpp

```diff
@@ -0,0 +1,49 @@
+#include <array>
+#include <cloysterhpc/tempdir.h>
+#include <fmt/format.h>
+#include <random>
+
+static char value_to_char(unsigned long value)
```

Comment:
```suggestion
static char value_to_char(std::uint64_t value)
```

This would make more sense because it express intent.